### ISSUE TITLE
Partially Initialized Model

### ISSUE THREAD

**Issue by jeffreyianwilson** [autodistill/autodistill#40]
```
from autodistill_grounded_sam import GroundedSAM
from autodistill_yolov8 import YOLOv8
from autodistill.detection.caption_ontology import CaptionOntology

# define an ontology to map class names to our GroundingDINO prompt
# the ontology dictionary has the format {caption: class}
# where caption is the prompt sent to the base model, and class is the label that will
# be saved for that caption in the generated annotations
base_model = GroundedSAM(ontology=CaptionOntology({"car": "car"}))

# label all images in a folder called `context_images`
base_model.label(
  input_folder="D:/work/dev/python/segmentation/test/03_split/",
  output_folder="D:/work/dev/python/segmentation/test/03_split/"
)

#target_model = YOLOv8("yolov8n.pt")
#target_model.train("./dataset/data.yaml", epochs=200)

# run inference on the new model
#pred = target_model.predict("./dataset/valid/your-image.jpg", confidence=0.5)
#print(pred)
```

Error
```
D:\work\dev>C:/Python/Python311/python.exe d:/work/dev/github/autodistill/autodistill.py
Traceback (most recent call last):
  File "d:\work\dev\github\autodistill\autodistill.py", line 1, in <module>
    from autodistill_grounded_sam import GroundedSAM
  File "C:\Python\Python311\Lib\site-packages\autodistill_grounded_sam\__init__.py", line 1, in <module>
    from autodistill_grounded_sam.grounded_sam import GroundedSAM
  File "C:\Python\Python311\Lib\site-packages\autodistill_grounded_sam\grounded_sam.py", line 17, in <module>
    from autodistill.detection import CaptionOntology, DetectionBaseModel
  File "d:\work\dev\github\autodistill\autodistill\detection\__init__.py", line 1, in <module>
    from autodistill.detection.caption_ontology import CaptionOntology
  File "d:\work\dev\github\autodistill\autodistill\detection\caption_ontology.py", line 4, in <module>
    from autodistill.detection import DetectionOntology
ImportError: cannot import name 'DetectionOntology' from partially initialized module 'autodistill.detection' (most likely due to a circular import) (d:\work\dev\github\autodistill\autodistill\detection\__init__.py)
```

**Comment by capjamesg** [comment]
Hello there! I have a PR in progress that should resolve this. I will message in this thread when the PR is done.

**Comment by capjamesg** [comment]
This should now be fixed in `autodistill==0.1.7`. Please test and let us know if you run into any issues!

This is a duplicate of #30 so I am closing. If you run into any issues, please respond in #30.
